### ISSUE TITLE
CRN https://pandoracrn4.crabbz.xyz/ without metrics

### ISSUE THREAD

**Issue by bloobsy** [aleph-im/support#1]
CRN has been checked and everything seems to be in perfect order according to Andres but the node is still receiving no metrics. It's been almost a week like this.

**Comment by hoh** [comment]
The URL of this node appears to not respond on IPv6.

To test, use either:

```
curl -6 -vi https://pandoracrn4.crabbz.xyz
```

Result:
```
* Host pandoracrn4.crabbz.xyz:443 was resolved.
* IPv6: 2a09:2dc0:0:a0::1
* IPv4: (none)
*   Trying [2a09:2dc0:0:a0::1]:443...
```
(hanging forver)

or check https://ready.chair6.net/?url=https%3A%2F%2Fpandoracrn4.crabbz.xyz -> IPv6 Connectivity

**Comment by hoh** [comment]
The `aleph-vm-supervisor` service and Caddy server in the installation procedure by default listen on both IPv4 and IPv6. This error must come from some configuration specific to your node.

**Comment by bloobsy** [comment]
ok, I've addressed the issue: 

root@blade48-1:~# curl -6 -vi https://pandoracrn4.crabbz.xyz
*   Trying [2a09:2dc0:0:a0::2]:443...
* Connected to pandoracrn4.crabbz.xyz (2a09:2dc0:0:a0::2) port 443 (#0)
* ALPN: offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=pandoracrn4.crabbz.xyz
*  start date: Sep 21 08:31:21 2024 GMT
*  expire date: Dec 20 08:31:20 2024 GMT
*  subjectAltName: host "pandoracrn4.crabbz.xyz" matched cert's "pandoracrn4.crabbz.xyz"
*  issuer: C=US; O=Let's Encrypt; CN=E5
*  SSL certificate verify ok.
* using HTTP/2
* h2h3 [:method: GET]
* h2h3 [:path: /]
* h2h3 [:scheme: https]
* h2h3 [:authority: pandoracrn4.crabbz.xyz]
* h2h3 [user-agent: curl/7.88.1]
* h2h3 [accept: */*]
* Using Stream ID: 1 (easy handle 0x55f53f8ea7f0)
> GET / HTTP/2
> Host: pandoracrn4.crabbz.xyz
> user-agent: curl/7.88.1
> accept: */*
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
< HTTP/2 200
HTTP/2 200
< alt-svc: h3=":443"; ma=2592000
alt-svc: h3=":443"; ma=2592000
< content-type: text/html
content-type: text/html
< date: Tue, 01 Oct 2024 11:55:47 GMT
date: Tue, 01 Oct 2024 11:55:47 GMT
< server: Caddy
server: Caddy
< server: aleph-vm/1.1.0
server: aleph-vm/1.1.0
< content-length: 15703
content-length: 15703

<
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Aleph.im Compute Node</title>
    <link rel="stylesheet" href="/static/main.css">
    <script src="/static/lightweight-charts.standalone.production.js"></script>
</head>
<body>
    <main>

        <h1>Aleph.im Compute Node</h1>

        <section>
            <p>
                This is an <a href="https://aleph.im/">Aleph.im</a> compute resource node.
            </p>
            <p>
                It executes user programs stored on the aleph.im network in Virtual Machines.
            </p>
            <p>
                See the <a href="https://github.com/aleph-im/aleph-vm">source code repository</a> for more info.
            </p>

        </section>

        <hr/>

        <section>
            <h2>Multiaddr</h2>
            <p>
                This node is exposed on the following addresses:
            </p>
            <ul>
                <li><a href="https://pandoracrn4.crabbz.xyz/">https://pandoracrn4.crabbz.xyz/</a></li>
                <li>/dns4/pandoracrn4.crabbz.xyz/tcp/443/https</li>
                <li>/dns6/pandoracrn4.crabbz.xyz/tcp/443/https</li>
            </ul>

        </section>

        <hr/>

        <section>
        <h2>Diagnostic</h2>

        <div id="virtualization-checks">
            <h3>Virtualization</h3>
            <p>
                Virtualization
                <span id="virtualization-check">
                ...
                    <span class="loader-container">
                        <span id="loader-one" class="loader"></span>
                        <span id="loader-two" class="loader"></span>
                        <span id="loader-three" class="loader"></span>
                    </span>
                </span>
            </p>
            <div class="details">
                <ul></ul>
            </div>
            <div class="help" style="display: none">
                <p>
                    ℹ️ <a href="https://docs.aleph.im/nodes/compute/troubleshooting/#virtualization">
                        Troubleshoot virtualization
                    </a>
                </p>
            </div>
        </div>

        <div id="virtualization-legacy-checks">
            <h3>Virtualization (legacy)</h3>
            <p>
                Virtualization
                <span id="virtualization-legacy-check">
                ...
                    <span class="loader-container">
                        <span id="loader-one" class="loader"></span>
                        <span id="loader-two" class="loader"></span>
                        <span id="loader-three" class="loader"></span>
                    </span>
                </span>
            </p>
            <div class="details">
                <ul></ul>
            </div>
            <div class="help" style="display: none">
                <p>
                    ℹ️ <a href="https://docs.aleph.im/nodes/compute/troubleshooting/#virtualization">
                        Troubleshoot virtualization
                    </a>
                </p>
            </div>
        </div>

        <div id="host-checks">
            <h3>Host connectivity</h3>
            <p>
                Host
                <span id="host-check">
                ...
                    <span class="loader-container">
                        <span id="loader-one" class="loader"></span>
                        <span id="loader-two" class="loader"></span>
                        <span id="loader-three" class="loader"></span>
                    </span>
                </span>
            </p>
            <div class="details">
                <h4>IPv4</h4>
                <ul class="ipv4"></ul>
                <h4>IPv6</h4>
                <ul class="ipv6"></ul>
            </div>
            <div class="help" style="display: none">
                <p>
                    ℹ️ <a href="https://docs.aleph.im/nodes/compute/troubleshooting/#host">
                        Troubleshoot host
                    </a>
                </p>
            </div>
        </div>

        <details>
            <summary>ℹ️ More information</summary>

            <div id="latest-metrics">
                <h3>Latest metrics</h3>
                <p>
                    The aleph.im network measures the performance of all nodes in the network. New metrics are published
                    every 10 minutes.
                </p>
                <p>
                    🔍 <a href="https://explorer.aleph.im/messages?showAdvancedFilters=1&channels=aleph-scoring&sender=0x4D52380D3191274a04846c89c069E6C3F2Ed94e4&page=1"
                        >Browse the metrics in the explorer</a>
                </p>
                <ul id="latest-metrics"></ul>
            </div>

            <h3>VM Egress IPv6</h3>
            <p>
                VM Egress IPv6 is a test to check if virtual machines are able to connect to the IPv6 internet.
                Enabling VM IPv6 Egress requires a specific configuration that is not applied automatically. It is not yet
                required to run virtual machines.
            </p>
            <div id="ipv6-egress-checks">
                <p>
                    VM Egress IPv6
                    <span id="ipv6-egress-check">
                        is ...
                        <span class="loader-container">
                            <span id="loader-one" class="loader"></span>
                            <span id="loader-two" class="loader"></span>
                            <span id="loader-three" class="loader"></span>
                        </span>
                    </span>
                </p>
            </div>
            <h3>APIs</h3>
            <p>
                Host status check API: <a href="/status/check/host">/status/check/host</a>
            </p>
            <p>

                Virtualization check API: <a href="/status/check/fastapi">/status/check/fastapi</a>
            </p>
            <p>

                VM Egress IPv6: <br/>
                <a href="/vm/63faf8b5db1cf8d965e6a464a0cb8062af8e7df131729e48738342d956f29ace/ip/6">/vm/63faf8b5db1cf8d965e6a464a0cb8062af8e7df131729e48738342d956f29ace/ip/6</a>
            </p>
        </details>

        </section>

        <section>
            <h2>Version</h2>
            <p>
                Running version <i>1.1.0</i>.
            </p>

            <div>
                <button id="status_check_button">Check if this is the latest release</button>
                <span id="status_latest_version" style="display: none">This is the latest release &#10004;&#65039;</span>
                <span id="status_outdated_version" style="display: none">
                    The latest release is <a href="https://github.com/aleph-im/aleph-vm/releases/latest"></a> &#10060;
                </span>
                <p id="status_error_version" style="display: none">Your browser could not fetch the latest release:
                    <span id="status_error_version_reason"></span>
                </p>
            </div>

            <div>
                <button id="metrics_chart_button">Load metrics chart</button>
                <p>This downloads about 35 MB of data from the network</p>
            </div>

            <div id="chart-wrapper">
                <div id="progress">
                    <span id="progress-bar-text">0%</span>
                    <progress id="progress-bar" value="0" max="100"></progress>
                </div>

                <div class="flex" id="checkboxes-wrapper"></div>

                <div id="chart"></div>

                <footer>
                    This chart is using <a href="https://github.com/tradingview/lightweight-charts">Lightweight Charts</a>, under the <a href="https://www.apache.org/licenses/LICENSE-2.0">Apache 2.0 license</a>.
                </footer>
            </div>
        </section>
    </main>


    <script src="/static/helpers.js"></script>
    <script>
        const NODE_HOSTNAME = 'https://pandoracrn4.crabbz.xyz/';
        const NODE_VERSION = '1.1.0';

        (async () => {
            try {
                const { status, details } = await fetchFastapiCheckStatus();
                document.getElementById('virtualization-check').innerHTML = status;
                if(Object.keys(details).length > 0){
                    const detailsDiv = document.querySelector("#virtualization-checks .details ul");
                    detailsDiv.innerHTML = objectToString(details);
                    document.querySelector("#virtualization-checks .help").style.display = "block";
                }
            } catch (err) {
                console.error('Could not fetch api status', err);
            }
        })();

        (async () => {
            try {
                const { status, details } = await fetchFastapiCheckStatus(legacy=true);
                document.getElementById('virtualization-legacy-check').innerHTML = status;
                if(Object.keys(details).length > 0){
                    const detailsDiv = document.querySelector("#virtualization-legacy-checks .details ul");
                    detailsDiv.innerHTML = objectToString(details);
                    document.querySelector("#virtualization-legacy-checks .help").style.display = "block";
                }
            } catch (err) {
                console.error('Could not fetch api status', err);
            }
        })();

        (async () => {
            try {
                const { status, details } = await fetchHostCheckStatus();
                document.getElementById('host-check').innerHTML = status;
                if(Object.keys(details).length > 0){
                    document.querySelector("#host-checks .details ul.ipv4").innerHTML = objectToString(details["ipv4"]);
                    document.querySelector("#host-checks .details ul.ipv6").innerHTML = objectToString(details["ipv6"]);
                    document.querySelector("#host-checks .help").style.display = "block";
                }
            } catch (err) {
                console.error('Could not fetch api status', err);
            }
        })();

        (async () => {
            document.getElementById('status_check_button').addEventListener('click', async () => {
                try{
                    const latestRelease = await isLatestRelease();
                    if(latestRelease === NODE_VERSION){
                        document.getElementById('status_latest_version').style.display = 'block';
                        document.getElementById('status_outdated_version').style.display = 'none';
                    }
                    else{
                        document.getElementById('status_latest_version').style.display = 'none';
                        const wrapperDiv = document.getElementById('status_outdated_version');
                        wrapperDiv.style.display = 'block';
                        wrapperDiv.querySelector('a').textContent = latestRelease;
                    }
                }
                catch(err){
                    document.getElementById('status_error_version').style.display = 'block';
                    document.getElementById('status_error_version_reason').innerHTML = err;
                }
            })
        })();

        (async () => {
            let data, chart

            const ONE_WEEK_AGO = Date.now() - 7 * 24 * 60 * 60 * 1000;

            const displayedMetrics = {
                base_latency: true,
                base_latency_ipv4: true,
                diagnostic_vm_latency: true,
                full_check_latency: true,
            }

            // Creates the chart
            // This function is called when pressing the "load metrics chart" button
            // and when toggling metrics on/off
            const createChart = (linesToDisplay) => {
                chart = LightweightCharts.createChart(
                    document.getElementById('chart'),
                    { timeScale: { timeVisible: true } }
                );
                const colors = {
                    base_latency: '#202030',
                    base_latency_ipv4: '#00916E',
                    diagnostic_vm_latency: '#A51C30',
                    full_check_latency: '#1B98E0',
                }
                for (const [k, v] of Object.entries(data)) {
                    if (!linesToDisplay[k])
                        continue;
                    const lineSeries = chart.addLineSeries({
                        color: colors[k],
                        title: k
                    });
                    lineSeries.setData(v.sort((a, b) => a.time - b.time));
                }
            }

            // Loads the latest node metrics when pressing the button
            document.getElementById('metrics_chart_button').addEventListener('click', async () => {
                document.getElementById('chart-wrapper').style.display = "block";
                document.getElementById('metrics_chart_button').disabled = true;
                try{
                    for await (const release of fetchLatestMetrics(NODE_HOSTNAME, ONE_WEEK_AGO, false)) {
                        data = release.data
                        document.getElementById('progress-bar').value = release.progress * 100;
                        document.getElementById('progress-bar-text').innerHTML = (release.progress * 100 | 0) + '%';
                    }
                    createChart(displayedMetrics);
                }
                catch(err){
                    console.error(err);
                    document.getElementById('chart-wrapper').innerHTML = err;
                }

                // Creates checkboxes to disable/enable metrics
                Object.keys(displayedMetrics).map(field => {
                    const wrapperDiv = document.createElement('div')
                    const cb = document.createElement('input')
                    cb.type = 'checkbox'
                    cb.checked = true
                    cb.id = 'toggle_' + field + '_checkbox';
                    const label = document.createElement('label')
                    label.htmlFor = cb.id
                    label.innerHTML = field.replace('_', ' ')
                    wrapperDiv.appendChild(cb)
                    wrapperDiv.appendChild(label)
                    document.getElementById('checkboxes-wrapper').appendChild(wrapperDiv)

                    cb.addEventListener('change', e => {
                        chart.remove()
                        displayedMetrics[field] = e.target.checked
                        createChart(displayedMetrics)
                    });
                })
            })

        })();

    (async () => {
        try{
            const response = await fetch('/vm/63faf8b5db1cf8d965e6a464a0cb8062af8e7df131729e48738342d956f29ace/ip/6');
            if (response.ok) {
                document.getElementById("ipv6-egress-check").innerHTML = "is working &#10004;&#65039;";
            }
            else if (response.status === 503) {
                document.getElementById("ipv6-egress-check").innerHTML = "fails to be tested &#10060; ";
            }
            else if (response.status === 500) {
                document.getElementById("ipv6-egress-check").innerHTML = "is not yet available &#9932;";
            }
            else {
                document.getElementById("ipv6-egress-check").innerText = response.status;
            }
        }
        catch(err){
            console.error(err);
            document.getElementById("ipv6-egress-check").innerHTML = "fails to be tested &#10060; ";
        }
    })();

    </script>
</body>
</html>
* Connection #0 to host pandoracrn4.crabbz.xyz left intact

**Comment by hoh** [comment]
Great, I will close the issue then.
